docs/concepts/architecture/master-node-communication.md

```diff
@@ -106,3 +110,79 @@ cluster (connecting to the ssh server listening on port 22) and passes all
 GCE network in which the cluster is running.
+
+
+
+## Configuring communication ports
+
+> A note on the difference between specifying 0.0.0.0 or 127.0.0.1 as a bind `--address`. *0.0.0.0* will bind to ALL interfaces on the host. *127.0.0.1* will ONLY bind to localhost.
+
+
+### kube-apiserver
+
+| Argument                       | Default           | Description                              |
+| ------------------------------ | ----------------- | ---------------------------------------- |
+| --advertise-address            | --bind-address    | The IP address on which to advertise the apiserver to members of the cluster. This address must be reachable by the rest of the cluster |
+| --bind-address                 | default interface | The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. |
+| --etcd-servers                 |                   | List of etcd servers to connect with (scheme://ip:port), comma separated. |
+| --insecure-bind-address        | 127.0.0.1         | The IP address on which to serve the --insecure-port |
+| --insecure-port                | 8080              | The port on which to serve unsecured, unauthenticated access. It is assumed that firewall rules are set up such that this port is not reachable from outside of the cluster and that port 443 on the cluster's public address is proxied to this port. This is performed by nginx in the default setup. |
+| --kubernetes-service-node-port |                   | If non-zero, the Kubernetes master service (which apiserver creates/maintains) will be of type NodePort, using this as the value of the port. If zero, the Kubernetes master service will be of type ClusterIP. |
+| --secure-port                  | 6443              | The port on which to serve HTTPS with authentication and authorization. Set to 0 to disable |
+| --service-node-port-range      | 30000-32767       | A port range to reserve for services with NodePort visibility. |
+
+
+| Argument     | Default                     | Description                              |
+| ------------ | --------------------------- | ---------------------------------------- |
+| --master     |                             | The address of the Kubernetes API server (overrides any value in kubeconfig) |
+| --kubeconfig | /var/lib/kubelet/kubeconfig | Path to a kubeconfig file, specifying how to connect to the API server. |
+
+### kube-controler-manager
+
+| Argument  | Default |
+| --------- | ------- |
+| --address | 0.0.0.0 |
+| --port    | 10252   |
+
+### cloud-controler-manager
+
+| Argument  | Default |
+| --------- | ------- |
+| --address | 0.0.0.0 |
+| --port    | 10253   |
+
+
+### kube-scheduler
+
+| Argument  | Default |
+| --------- | ------- |
+| --address |         |
+| --port    | 10251   |
+
+
+### kubelet
+
+> Note that kubelet does not support the `--master` flag
```

Comment:
Seems like this is redundant here. If we want to keep it, maybe we want to tell that which other components support this flag.